pkg/apiclient/apiclient.go

```diff
@@ -67,7 +69,11 @@ func NewClientFromOpts(opts Opts) (context.Context, Client, error) {
 			opts.ClientConfig = opts.ClientConfigSupplier()
 		}
 
-		ctx, client, err := newArgoKubeClient(opts.ClientConfig, instanceid.NewService(opts.InstanceID))
+		if opts.Context == nil {
```

Comment:
minor - you could add `opts.GetContext()`

Reply:
Could you elaborate more on this? Do you mean moving also the `opts.Context = context.Background()` to this `GetContext` method, so I creates a context, if none was provided?

Reply:
just a convenience

Reply:
Done